src/LanguageServer/Impl/LanguageServer.cs

```diff
@@ -168,7 +169,17 @@ private void OnUnregisterCapability(object sender, UnregisterCapabilityEventArgs
                 await _server.DidChangeConfiguration(new DidChangeConfigurationParams { settings = settings }, cancellationToken);
 
                 if (!_filesLoaded) {
-                    await LoadDirectoryFiles();
+                    try {
+                        await LoadDirectoryFiles();
+                    } catch (Exception ex) when (
+                        ex is IOException // FileNotFoundException, DirectoryNotFoundException, PathTooLongException, etc
+                        || ex is UnauthorizedAccessException
+                    ) {
+                        // These exceptions are not caused by the LS, but by invalid/inaccessible user-specified paths.
+                        _server.LogMessage(MessageType.Error, $"Failed to load files in {_initParams.rootPath} - {ex.GetType().Name}: {ex.Message}");
+                        // TODO: Should this exit completely, or stay open and give the user a chance to fix their config?
```

Comment:
I think staying open is OK. Some conflict with, say, virus scanner, should not cause crashes.

Reply:
I think the answer is probably "no", but should that `LogMessage` instead be a `ShowMessage` to produce a more visible indication that something's wrong? Before, it'd crash, but now it'll (somewhat) silently sit there having not loaded any files.

Reply:
We can upgrade the error to `ShowMessage` to give user stronger hint about the problem.

Reply:
Upgraded to `ShowMessage` and removed the TODO.